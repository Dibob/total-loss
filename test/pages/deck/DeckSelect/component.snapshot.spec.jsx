import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter, Link } from 'react-router-dom';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DeckPageComponent from '../../../../src/app/pages/deck/DeckSelect/component';

describe('DeckList page', () => {
  const deckList = [
    {
      id: 1,
      name: 'Testing card id 1',
      description: 'Some testing card',
    },
    {
      id: 2,
      name: 'Testing card id 2',
      description: 'Some testing card',
    },
    {
      id: 3,
      name: 'Testing card id 3',
      description: 'Some testing card',
    },
  ];

  beforeEach(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  it('renders DeckList page component', () => {
    const rendered = renderer.create(
      <MemoryRouter>
        <DeckPageComponent decks={deckList} selectDeck={() => mockFunc()} />
      </MemoryRouter>,
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should call selectDeck event', () => {
    const mockFunc = jest.fn();

    const pageWrapper = mount(
      <MemoryRouter>
        <DeckPageComponent decks={deckList} selectDeck={deckId => mockFunc(deckId)} />
      </MemoryRouter>,
    );

    pageWrapper
      .find(Link)
      .at(0)
      .simulate('click');

    expect(mockFunc.mock.calls.length).toBe(1);
  });
});
