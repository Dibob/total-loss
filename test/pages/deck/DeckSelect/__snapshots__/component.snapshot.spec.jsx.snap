// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckList page renders DeckList page component 1`] = `null`;
